tests/subscriptions/test_worker.py

```diff
@@ -0,0 +1,127 @@
+from concurrent.futures import ThreadPoolExecutor
+from datetime import datetime, timedelta
+from typing import Iterable, Iterator
+from uuid import uuid1
+
+from snuba.subscriptions.consumer import Tick
+from snuba.subscriptions.data import (
+    PartitionId,
+    Subscription,
+    SubscriptionData,
+    SubscriptionIdentifier,
+)
+from snuba.subscriptions.scheduler import ScheduledTask, Scheduler
+from snuba.subscriptions.worker import SubscriptionTaskResult, SubscriptionWorker
+from snuba.utils.metrics.backends.dummy import DummyMetricsBackend
+from snuba.utils.streams.consumer import Consumer
+from snuba.utils.streams.dummy import DummyBroker, DummyConsumer, DummyProducer
+from snuba.utils.streams.types import Message, Partition, Topic
+from snuba.utils.types import Interval
+from tests.subscriptions import BaseSubscriptionTest
+
+
+class DummyScheduler(Scheduler[Subscription]):
+    def __init__(self, tasks: Iterable[ScheduledTask[Subscription]]):
+        self.__tasks = tasks
+
+    def find(
+        self, interval: Interval[datetime]
+    ) -> Iterator[ScheduledTask[Subscription]]:
+        for task in self.__tasks:
+            yield task
+
+
+class TestSubscriptionWorker(BaseSubscriptionTest):
+    def test(self):
+        result_topic = Topic("subscription-results")
+
+        broker: DummyBroker[SubscriptionTaskResult] = DummyBroker()
+        broker.create_topic(result_topic, partitions=1)
+
+        subscription = Subscription(
+            SubscriptionIdentifier(PartitionId(0), uuid1()),
+            SubscriptionData(
+                project_id=self.project_id,
+                conditions=[["platform", "IN", ["a"]]],
+                aggregations=[["count()", "", "count"]],
+                time_window=timedelta(minutes=500),
+                resolution=timedelta(minutes=1),
+            ),
+        )
+
+        now = datetime.utcnow()
+
+        timestamps = [
+            now,
+            (now + timedelta(minutes=self.minutes) + subscription.data.time_window),
+        ]
+
+        scheduler = DummyScheduler(
+            ScheduledTask(timestamp, subscription) for timestamp in timestamps
+        )
+
+        worker = SubscriptionWorker(
+            self.dataset,
+            ThreadPoolExecutor(),
+            {0: scheduler},
+            DummyProducer(broker),
+            result_topic,
+            DummyMetricsBackend(strict=True),
+        )
```

Comment:
Nothing actually covered the `SubscriptionWorker` before so I thought it made sense to turn the existing executor test into a `SubscriptionWorker` test, which itself became almost an integration test against the dummy Kafka cluster, with the exception that it doesn't use the real `TickConsumer`.